packages/flutter_tools/lib/src/platform_plugins.dart

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Marker interface for all platform specific plugin config impls.
+abstract class PlatformPlugin {
+  const PlatformPlugin();
+
+  Map<String, dynamic> toMap();
+}
+
+/// Contains all the required parameters to template an Android plugin.
+class AndroidPlugin extends PlatformPlugin {
```

Comment:
This seems like a better approach for splitting up the plugins. It would also be helpful to add some sort of unknown/unspecified platform that handles keys we don't recognize.